src/include/detail/ivf/qv.h

```diff
@@ -188,14 +102,19 @@ auto qv_query_heap_infinite_ram(
  * @return The indices of the top_k neighbors for each query vector
  */
 // @todo We should still order the queries so partitions are searched in order
+template <class U>
 auto qv_query_heap_infinite_ram(
+    U,
     auto&& partitioned_db,
     auto&& centroids,
     auto&& q,
     auto&& indices,
     auto&& partitioned_ids,
     size_t nprobe,
     size_t k_nn,
+    auto&& updates,
+    auto&& update_ids,
+    auto&& update_validity,
```

Comment:
We can use a more intuitive set of arguments:
```
    auto&& updates,
    auto&& update_ids,
    auto&& delete_ids,
```

In this way we also don't need the code that splits updates into deletes and inserts. The filtering of the ivf results will need to happen using a union of delete_ids and update_ids. And the flat query can take all update vectors as input.

Reply:
To avoid copying data from Python we can also add some methods that read updates/deletes from an array uri. We can use the same schema as the one defined in https://github.com/TileDB-Inc/TileDB-Vector-Search/pull/128/files#diff-b0ba694c23a77c9684b7d183ae060fd615e208a54123ed3d384a3aaf9d30b214R126
In this array schema, deletes are represented by empty vector attributes.